db/common/src/main/java/io/helidon/db/common/MappingProcessor.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.db.common;
+
+import java.util.function.Function;
+
+import io.helidon.common.reactive.Flow;
+
+/**
+ * A {@link Flow.Processor} that only maps the source type to target type using a mapping function.
+ *
+ * @param <SOURCE> type of the publisher we subscribe to
+ * @param <TARGET> type of the publisher we expose
+ */
+public final class MappingProcessor<SOURCE, TARGET> implements Flow.Processor<SOURCE, TARGET> {
+    private final Function<SOURCE, TARGET> resultMapper;
```

Comment:
No Mapper<A,B> here?

Reply:
The resultMapper may come from various places, `Mapper<A, B>` is just one of them - it may be from `DbMapper`, or an identity function.